src/main/java/stsjorbsmod/cards/cull/Siphon.java

```diff
@@ -22,19 +22,22 @@
     private static final int UPGRADE_DAMAGE = 3;
     private static final int HEAL_ON_FATAL = 4;
     private static final int UPGRADE_PLUS_HEAL_ON_FATAL = 1;
-
+    private static final int MAX_HP_ON_FATAL = 1;
 
     public Siphon() {
         super(ID, COST, TYPE, COLOR, RARITY, TARGET);
         baseDamage = DAMAGE;
+        metaMagicNumber = baseMetaMagicNumber = MAX_HP_ON_FATAL;
         magicNumber = baseMagicNumber = HEAL_ON_FATAL;
         tags.add(CardTags.HEALING);
     }
 
     @Override
     public void use(AbstractPlayer p, AbstractMonster m) {
-        Runnable onKillEffect = () -> addToBot(new HealAction(p, p, magicNumber));
-        this.addToBot(new DamageWithOnKillEffectAction(m, new DamageInfo(p, this.damage, this.damageTypeForTurn), onKillEffect, true));
+        Runnable onKillEffect = () -> addToBot(
+                new IncreaseMaxHpByFlatAmountActionAndHealAction(p, p, magicNumber, metaMagicNumber, true));
```

Comment:
Why did we choose to create an action instead of 
```java
() -> {
  addToBot(maxHpAction);
  addToBot(healAction);
}
```
brain might be farting a bit.

Reply:
I think the answer is mostly because I didn't think of it, let me try it out and see if it works as expected.

Reply:
Yep, works and is definitely a better solution, thanks!